### DISCUSSION TITLE
Using STM32 internal ADC with interrupt:

### DISCUSSION THREAD

**Discussion by mtahirbutt** [zephyrproject-rtos/zephyr#44751] [Q&A]
Hi, I am trying to use internal ADC from STM32 for disco_l475_iot1 board, I tried to use struct adc_sequence_options but I am not sure if this is the right thing to do for interrupt. 

Someone could help me in this regard. Thanks in advance.

**Comment by FRASTM** [comment]
In the stm32l475x deivece, ADC1 & 2 & 3 have internal channels : 
ADC1 is connected to 16 external channels + 3 internal channels  --> channels 0, 17, 18
ADC2 is connected to 16 external channels + 2 internal channels --> channels 17, 18
ADC3 is connected to 12 external channels + 4 internal channels --> channels 14, 15, 17, 18 

They are assigned to internal Vref or Vsensor temp or Vbat or DAC 1/ 2

See https://github.com/zephyrproject-rtos/zephyr/pull/43861 as a example for deifining/using one of the ADC internal channels.

**Reply by mtahirbutt** [reply]
thanks sir for the reply. but I am able to use ADC with adc_sequence_options using callback for sampling event. but still working on using IRQ or ISR (using interrupt) for ADC. As using interrupt might result in strict timing with minimum jitter. anyways thanks a lot for the help.

**Reply by Parinaz-Motus** [reply]
Hi there, 
could you figure how to do it? I added a new property called trigger in the adc yaml file and used this function LL_ADC_REG_SetTriggerSource(adc, (config->trigger)<<5) to set the trigger source. I'm now struggling with the callback.
